src/apiServices/AuthService.ts

```diff
@@ -33,6 +33,21 @@ export class AuthService {
         }
     }
 
+    public async loginWithMetamask(signature: string): Promise<JwtTokensInterface | void> {
```

Comment:
Can we rename this something like `loginWithSignature`, as it doesn't actually use metamask?

Can we also create a different `loginWithWeb3` method that actually uses metamask?
So the flow of the `loginWithWeb3` function could be:

```
if (typeof window.ethereum !== 'undefined'
|| (typeof window.web3 !== 'undefined')) {
  // Web3 browser user detected. You can now use the provider.
  const provider = window['ethereum'] || window.web3.currentProvider
} else { Error }

// Do whatever checks to make sure the account is available in metamask

chal = getChallenge(provider.selectedAddress)
signature = web3.personalSign(chal)
loginWithMetamask(signature)
```

Reply:
I was looking at this with today, apparently for some reason the in certain browser signature does not work with `window.web3`, but you have to install and import the `Web3` library. The reason why I didnt do it the way you mentioned above is because I do not want to import that massive library (200kb+).

Another reason is that we dont want (and we shouldnt) support all the different Metamask versions (for example Joao is not using latest version of MM on his dashboard because he was having issues with it). If there is some change in MM we should take care of, but also not break older versions etc. so it can get tricky easily.

Thats why I preferred to just expose those two methods and let the user handle the Metamask login themself.

If we think this is really beneficial we can implement it ourself, even though I think it makes more sense to have the user do it on the front end due to reasons above. Below is how I implemented in a react app I have in my machine:

` return new Promise((resolve, reject) => {
            const {value} = data.message
            if (!account) {
              console.error('no-account')
            }
              const params = [account, JSON.stringify(data)]
              console.log('paramooo', params)
            //   const web3 = getWeb3()/
              console.log(account, 'account')
              web3.currentProvider.sendAsync(
                        { method: 'eth_signTypedData_v3', params, from: account },
                        (err, result) => {
                          if (err || result.error) {
                            console.error('THERE WAS AN ERRORRRR', err)
                          }
                          resolve(value + '.' + result.result)
                        }
                    )
            }).catch((error) => {
              // errorHandler(error)
              console.error(error)
            })`

Reply:
If we do think this makes more sense to have in the library I can implement it..

Reply:
Fair enough. We should probably eventually provide example code for this feature in the docs though.

I still think we can rename this method to `loginWithSignature` though.